### PR TITLE
Add option to r(x)dock to upload user-defined receptor.prm files

### PR DESCRIPTION
**Pull request by simonbray**
@bgruening, @tdudgeon, here's a really simple implementation

### CODE REVIEW

File: chemicaltoolbox/rdock/rbdock.xml [hunk 1]
```diff
@@ -7,7 +7,15 @@
     <command><![CDATA[
 ln -s '$active_site' receptor.as &&
 ln -s '$receptor' receptor.mol2 &&
-ln -s $receptor_prm receptor.prm &&
+
+#if $custom_rec.custom_rec_select == 'true':
+    ## simple check that the receptor file name is correct
+    grep -q 'RECEPTOR_FILE receptor.mol2' $custom_rec.prm_file &&
```
Comment (simonbray) [inline chemicaltoolbox/rdock/rbdock.xml:13]: only thing that could be improved is to provide a proper error message, not sure how to do this in a simple way. But the user is warned clearly in the help for the file input

Comment (bgruening) [reply chemicaltoolbox/rdock/rbdock.xml:13]: should we fail or straight replace?

File: chemicaltoolbox/rdock/rbdock.xml [hunk 2]
```diff
@@ -42,14 +50,29 @@ cat rdock_output.sd
...
         </configfile>
     </configfiles>
 
     <inputs>
         <param type="data" name="receptor" format="mol2" label="Receptor" help="Select a receptor (mol2 format)."/>
         <param type="data" name="active_site" format="rdock_as" label="Active site" help="Active site file"/>
         <param type="data" name="ligands" format="sdf,mol" label="Ligands" help="Ligands in SDF format (or single ligand in MOL format)"/>
+        <conditional name="custom_rec">
+            <param name="custom_rec_select" type="select" label="Use a custom user-defined receptor.prm file?" help="Useful for more advanced use-cases, such as tethered docking.">
+                    <option value="true">Use custom receptor.prm file</option>
```
Comment (bgruening) [inline chemicaltoolbox/rdock/rbdock.xml:65]: indentation ..

File: chemicaltoolbox/rdock/rbdock.xml [hunk 3]
```diff
@@ -42,14 +50,29 @@ cat rdock_output.sd
...
+#end if
         </configfile>
     </configfiles>
 
     <inputs>
         <param type="data" name="receptor" format="mol2" label="Receptor" help="Select a receptor (mol2 format)."/>
         <param type="data" name="active_site" format="rdock_as" label="Active site" help="Active site file"/>
         <param type="data" name="ligands" format="sdf,mol" label="Ligands" help="Ligands in SDF format (or single ligand in MOL format)"/>
+        <conditional name="custom_rec">
+            <param name="custom_rec_select" type="select" label="Use a custom user-defined receptor.prm file?" help="Useful for more advanced use-cases, such as tethered docking.">
```
Comment (bgruening) [inline chemicaltoolbox/rdock/rbdock.xml:64]: Use a custom user-defined receptor.prm file? --> Use a custom-defined receptor.prm file?

File: chemicaltoolbox/rdock/rbdock.xml (hunk 1 above)
Comment (bgruening) [reply chemicaltoolbox/rdock/rbdock.xml:13]: I guess something like that could work `grep "regex" "filepath" || echo "no match" >&2`

File: chemicaltoolbox/rdock/rbdock.xml [hunk 4]
```diff
@@ -42,14 +50,29 @@ cat rdock_output.sd
...
+                    <option value="true">Use custom receptor.prm file</option>
+                    <option value="false" selected="true">Use default options (no user upload required)</option>
+            </param>
+            <when value="true">
+                <param type="data" name="prm_file" format="txt" label="Receptor prm file" help="Note the file must contain the line 'RECEPTOR_FILE receptor.mol2', else the tool will fail."/>
+            </when>
+            <when value="false">
+                <param name="flex" type="float" value="3.0" label="Receptor flex" help="Value for RECEPTOR_FLEX (default 3)"/>
```
Comment (bgruening) [inline chemicaltoolbox/rdock/rbdock.xml:72]: min="0.0"?

File: chemicaltoolbox/rdock/rbdock.xml (hunk 1 above)
Comment (simonbray) [reply chemicaltoolbox/rdock/rbdock.xml:13]: I prefer fail so the user is informed something was wrong

Comment (bgruening) [reply chemicaltoolbox/rdock/rbdock.xml:13]: fair enough!

**Comment by bgruening** [comment]
Gracias!

**Comment by bgruening** [comment]
installed.

**Pull request merged**